RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -366,6 +366,16 @@ class PumpOpsSynchronous {
                             events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), atIndex: 0)
                         }
                     }
+                    
+                    if let alarm = event as? PumpAlarmPumpEvent {
+                        switch alarm.alarmType {
+                        case .BatteryDepleted, .BatteryOutLimitExceeded, .DeviceReset:
+                            print("Found clock loss in pump history.  Ending history fetch.")
```

Comment:
So the assumption is that all events prior to the battery out are properly
timestamped until an even older changeTime event is found, which continues
the offset management.

I may never have to use my crummy 522 with the errors; if I do, I can edit
this code to be more resilient to battery changes.
On Thu, Jul 28, 2016 at 1:17 PM Pete Schwamb <notifications@github.com>
wrote:

> In RileyLinkKit/PumpOpsSynchronous.swift
> <https://github.com/ps2/rileylink_ios/pull/174#discussion_r72694306>:
>
> > @@ -366,6 +366,16 @@ class PumpOpsSynchronous {
> >                              events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), atIndex: 0)
> >                          }
> >                      }
> > +
> > +                    if let alarm = event as? PumpAlarmPumpEvent {
> > +                        switch alarm.alarmType {
> > +                        case .BatteryDepleted, .BatteryOutLimitExceeded, .DeviceReset:
> > +                            print("Found clock loss in pump history.  Ending history fetch.")
>
> Yeah, I was wondering if that's what happened with @loudestnoise
> <https://github.com/loudestnoise>'s pump.
>
> The reset to zero is an interesting idea, but wouldn't it cause bad
> interpretation of older events if there was also a valid time change
> interval?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ps2/rileylink_ios/pull/174/files/978a5313464917dac8f9aba55f0355569f55b82c#r72694306>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM_jMGesD6FoOIiOqyVT8HWOU4_zEhLks5qaQ5XgaJpZM4JW2jr>
> .
>